### ISSUE TITLE
distro does not expose minor version on CentOS 7.3

### ISSUE THREAD

**Issue by tlandschoff-scale** [nir0s/distro#212]
Hello world,

we are using distro in our builds for a while now instead of hand-rolled distribution version detection. Thanks for helping us out!

A build for CentOS 7.3 was incomplete lately and it turns out that distro does not seem to correctly identify the minor version:

```
horatio:~$ sudo docker run -it centos:7
[root@2590b53eef75 /]# yum install --quiet -e 0 -y python
[...]
[root@2590b53eef75 /]# curl --silent https://bootstrap.pypa.io/get-pip.py | python /dev/stdin --quiet
[root@2590b53eef75 /]# pip install distro
Collecting distro
  Downloading https://files.pythonhosted.org/packages/c1/e4/933159b5f7f9f5b7ae463e76f58da84a30d0943ab3c162c366a9ad95f01e/distro-1.2.0-py2.py3-none-any.whl
Installing collected packages: distro
Successfully installed distro-1.2.0
[root@2590b53eef75 /]# python
Python 2.7.5 (default, Aug  4 2017, 00:39:18) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-16)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import distro
>>> distro.id(), distro.version(), distro.version_parts()
(u'centos', u'7', (u'7', '', ''))
>>> distro.version(best=True)
u'7.3.1611'
>>> exit()
[root@2590b53eef75 /]# cat /etc/redhat-release
CentOS Linux release 7.3.1611 (Core) 
```

I just noticed that `best=False` is actually the default and should fix our build. However, after `distro.version()` returns the minor version just fine on CentOS 6.9 I wonder if others will run into this.

Basically I would have expected that the amount of information in `distro.version()` to be stable across releases of a given distribution.

Feel free to close this immediately, I am just wondering if `best=True` would be a better default :-)

Greetings, Torsten

**Comment by nir0s** [comment]
The reason why distro doesn't default to the best version is because it favors consistency over precision. Centos 7 versions only provide "7" as their version number in `/etc/os-release` but provide the exact version in another file distro deems lower priority.

To answer your question, the problem isn't with distro, but rather with the different distributions being "clever".

Having said that, there's an effort to provide a distro inconsistencies framework, which I hope we'll introduce soon. There, we can take care of weird quirks.

**Comment by tlandschoff-scale** [comment]
Thanks for your reply and consideration, much appreciated!
